pkg/routeagent_driver/handlers/healthchecker/healthchecker.go

```
@@ -0,0 +1,331 @@
+		if pingerObject.GetIP() == endpoint.Spec.HealthCheckIP {
+			logger.Infof("HealthChecker did not change %q ", endpoint.Name)
+			return nil
+		}
+
+		logger.Infof("HealthChecker is already running for %q - stopping", endpoint.Name)
+		pingerObject.Stop()
+		delete(h.pingers, endpoint.Spec.CableName)
+	}
+
+	pingerConfig := pinger.Config{
+		IP:                 endpoint.Spec.HealthCheckIP,
+		MaxPacketLossCount: h.config.MaxPacketLossCount,
+	}
+
+	if h.config.PingInterval != 0 {
+		pingerConfig.Interval = time.Second * time.Duration(h.config.PingInterval)
+	}
+
+	if h.config.MaxPacketLossCount != 0 {
+		pingerConfig.MaxPacketLossCount = h.config.MaxPacketLossCount
+	}
+
+	newPingerFunc := h.config.NewPinger
+	if newPingerFunc == nil {
+		newPingerFunc = pinger.NewPinger
+	}
+
+	pingerObject := newPingerFunc(pingerConfig)
+	h.pingers[endpoint.Spec.CableName] = pingerObject
+	pingerObject.Start()
+
+	logger.Infof("CableEngine HealthChecker started pinger for CableName: %q with HealthCheckIP %q",
+		endpoint.Spec.CableName, endpoint.Spec.HealthCheckIP)
+
+	return nil
+}
+
+func (h *controller) RemoteEndpointDeleted(endpoint *submarinerv1.Endpoint) error {
+	h.Lock()
+	defer h.Unlock()
+
+	if pingerObject, found := h.pingers[endpoint.Spec.CableName]; found {
+		pingerObject.Stop()
+		delete(h.pingers, endpoint.Spec.CableName)
+	}
+
+	delete(h.remoteEndpoint, endpoint.Name)
+
+	return nil
+}
+
+func (h *controller) Init() error {
+	go func() {
+		wait.Until(h.syncRouteAgentStatus, RouteAgentUpdateInterval, h.stopCh)
+	}()
+
+	return nil
+}
+
+// TransitionToNonGateway is called once for each transition of the local node from Gateway to a non-Gateway.
+func (h *controller) TransitionToNonGateway() error {
+	if h.config.HealthCheckerEnabled {
+		for i := range h.remoteEndpoint {
+			err := h.processEndpointCreatedOrUpdated(h.remoteEndpoint[i])
+			if err != nil {
+				logger.Warningf("Error processing remote endpoint %s: %v", h.remoteEndpoint[i].Name, err)
+			}
```

Perhaps we should return the error so it's retried?

---

Done.

---

This wasn't addressed.

---

Oh sorry,  Did an undo it between and it got reverted, fixed now.